extras/ioc_hash_scan.sh

```diff
@@ -0,0 +1,189 @@
+}
+
+function installNC() {
+  if [[ "$OS" == "Debian" ]]; then
+    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+      aws --profile="$AWSPROF" --region="$AWS_REGION" \
+      ssm start-session --target="$i" \
+      --document=AWS-StartInteractiveCommand \
+      --parameters="command='sudo apt-get install netcat -y'"
+  else
+    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+      aws --profile="$AWSPROF" --region="$AWS_REGION" \
+      ssm start-session --target="$i" \
+      --document=AWS-StartInteractiveCommand \
+      --parameters="command='sudo yum install netcat -y'"
+  fi
+}
+
+function startListen() {
+  if [[ "$OS" == "Debian" ]]; then
+    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+      aws --profile="$AWSPROF" --region="$AWS_REGION" \
+      ssm start-session --target="$i" \
+      --document=AWS-StartInteractiveCommand \
+      --parameters="command='cd ~/src/ioc_scan; nc -l -p 6666 | tar xzf -'"
+  else
+    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+      aws --profile="$AWSPROF" --region="$AWS_REGION" \
+      ssm start-session --target="$i" \
+      --document=AWS-StartInteractiveCommand \
+      --parameters="command='cd ~/src/ioc_scan; nc -l 6666 | tar xzf -'"
+  fi
+}
+
+## MAIN SCRIPT
+echo "IOC Hash Scan - $today-$(date +%H:%M:%S)" > "$logfile"
+
+for i in "${serverList[@]}"; do
+  OS="$(getOSType)"
+  if [[ "$OS" != "Debian" && "$OS" != "Fedora" ]]; then
+    echo "Non-supported OS Type"
+    exit 1
+  fi
+
+  instanceName=$(getHost)
+
+  echo "Begining scan of Instance $i -- $instanceName" | tee -a "$logfile"
+
+  # Start Port Forwarding
+  echo "Begining port forwarding"
+  portForward &
+
+  # Create ~/src/ioc_scan directory on Instance
+  echo "Verifying ~/src/ioc_scan directory on $instanceName"
+  AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+    aws --profile="$AWSPROF" --region="$AWS_REGION" \
+    ssm start-session --target="$i" \
+    --document=AWS-StartInteractiveCommand \
+    --parameters="command='if [ ! -d ~/src/ioc_scan ]; then mkdir -p ~/src/ioc_scan; fi'"
+
+  #Install netcat and start listening on port 6666
+  echo "Verifying netcat on $instanceName"
+  installNC
+  echo "Begin listening on $instanceName"
+  startListen &
+
+  # Copy latest ioc_scanner.py to target instance
+  curdir=$(pwd)
+  cd "$pydir" || exit
+
+  echo "Upload lastest ioc_scanner.py to $instanceName"
+  tar czf - ./ioc_scanner.py | nc localhost 5555
+
+  cd "$curdir" || exit
+
+  # Run ioc_scanner.py on target instance
+  echo "Scan $instanceName for IOC Hashes"
+  AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+    aws --profile="$AWSPROF" --region="$AWS_REGION" \
+    ssm start-session --target="$i" \
+    --document=AWS-StartInteractiveCommand \
+    --parameters="command=python3 ~/src/ioc_scan/ioc_scanner.py" >> "$logfile"
+
+  # Killing port forwading so we can do this again on the next Instance.
+  while pgrep -fq session-manager-plugin; do
```

Comment:
Did you try using [`aws ssm terminate-session`](https://docs.aws.amazon.com/cli/latest/reference/ssm/terminate-session.html) to stop the port-forwarding session?  That seems cleaner than killing the `session-manager-plugin`.

Reply:
`TerminateSession` is not allowed by the `StartStopSSMSession` role

Reply:
@dav3r - This seems like a permission we should add to `StartStopSSMSession`.  What do you think?

Reply:
Aha, here's the issue.  Since we are using an assumed role, the resource in question looks like this:
```
arn:aws:sts::${local.this_account_id}:assumed-role/StartStopSSMSession/&{aws:username}-*
```
and not like this (as currently defined in the policy mentioned above):
```
arn:aws:ssm:${var.aws_region}:${local.this_account_id}:session/&{aws:username}-*
```

Reply:
For additional troubleshooting/brainstorming, here is the session that @bra1ncramp attempted to terminate and failed:
```
arn:aws:ssm:us-east-1:<acct-number-redacted>:session/<username-redacted>-0fac6447f6dfdabf9 
```

Here is the full (sanitized) error message that @bra1ncramp sent to me:
```
An error occurred (AccessDeniedException) when calling the TerminateSession operation: 
User: arn:aws:sts::<acct-number-redacted>:assumed-role/StartStopSSMSession/<username-redacted>
is not authorized to perform: ssm:TerminateSession on resource: 
arn:aws:ssm:us-east-1:<acct-number-redacted>:session/<username-redacted>-0fac6447f6dfdabf9
because no identity-based policy allows the ssm:TerminateSession action
```

Reply:
@bra1ncramp - Where are you testing this?  Can you try changing
```hcl
  # Allow the user to terminate his or her own sessions
  statement {
    actions = [
      "ssm:TerminateSession",
    ]
    resources = [
      "arn:aws:ssm:${var.aws_region}:${local.this_account_id}:session/&{aws:username}-*",
    ]
  }
```
to
```hcl
  # Allow the user to terminate his or her own sessions
  statement {
    actions = [
      "ssm:TerminateSession",
    ]
    resources = [
      "arn:aws:ssm:${var.aws_region}:${local.this_account_id}:session/${aws:username}-*",
    ]
  }
```
in the policy `StartStopSSMSession`?  (Note that the only change is that the ampersand (`&`) is changed to a dollar sign (`$`) toward the end of the resource ARN.)  You can simply make this change live, in the AWS console for now.

Reply:
Nevermind, @bra1ncramp.  I was wrong.

Reply:
I created cisagov/session-manager-tf-module#29, which should resolve the issue and allow users to terminate their own sessions.

Reply:
Pending merge of https://github.com/cisagov/session-manager-tf-module/pull/29 before I can test.

Reply:
Once this last item gets resolved, I think this PR should be done.  @bra1ncramp please let us know when you've tested this (or if you have any blockers to your testing that we can help with).

Reply:
You may have been waiting on me for this. I have tested it and it is working. I was able to use it successfully, yesterday.

Reply:
Huzzah - glorious news! 🎉